docs/best-practices/testing.md

```diff
@@ -0,0 +1,32 @@
+# Testing best practices
+
+## Polling
+If you need to poll to wait for some event to happen, avoid using directly the `wait.PollImmediate` function: Kass Fleet Manager provides a rich set of utilities that will handle for you all the caveats of polling when dealing with it.
+
+Such utilities are:
+* **The poller builder**. A builder interface that will guide you in configuring a custom poller
+* **Cluster pollers**. A number of preconfigured pollers to be used to poll for a specific cluster event:
+    * **WaitForClustersMatchCriteriaToBeGivenCount**. Awaits for the number of clusters with an assigned cluster id to be exactly `count`
+    * **WaitForClusterIDToBeAssigned**. Awaits for clusterID to be assigned to the designed cluster
+    * **WaitForClusterToBeDeleted**. Awaits for the specified cluster to be deleted
+    * **WaitForClusterStatus**. Awaits for the cluster to reach the desired status
+* **Kafka pollers**. A number of preconfigured pollers that will allow to easily poll for a specific kafka event:
+    * **WaitForNumberOfKafkaToBeGivenCount**. Awaits for the number of kafkas to be exactly X
+    * **WaitForKafkaCreateToBeAccepted**. Creates a kafka and awaits for the request to be accepted
```

Comment:
I am wondering if this poller is still required, since right now `accepted` state is assigned during creation in the `RegisterKafkaJob` function

Reply:
I believe yes. It is still used in many tests:
* https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/11136041f9395eeefaa07a66803094ed3a61204d/internal/kafka/test/integration/kafkas_test.go#L103
* https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/11136041f9395eeefaa07a66803094ed3a61204d/internal/kafka/test/integration/kafkas_test.go#L1375
* https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/11136041f9395eeefaa07a66803094ed3a61204d/internal/kafka/test/integration/kafkas_test.go#L1394
* https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/11136041f9395eeefaa07a66803094ed3a61204d/internal/kafka/test/integration/kafkas_test.go#L1416

Reply:
ok. thanks